src/approaches/probe_approach.py

```diff
@@ -0,0 +1,201 @@
+"""Open-loop large language model (LLM) meta-controller approach.
+
+Example command line:
+    export OPENAI_API_KEY=<your API key>
+    python src/main.py --approach probe --seed 0 \
+        --strips_learner oracle \
+        --env pddl_blocks_procedural_tasks \
+        --num_train_tasks 3 \
+        --num_test_tasks 1 \
+        --debug
+
+Easier setting:
+    python src/main.py --approach probe --seed 0 \
+        --strips_learner oracle \
+        --env pddl_easy_delivery_procedural_tasks \
+        --pddl_easy_delivery_procedural_train_min_num_locs 2 \
+        --pddl_easy_delivery_procedural_train_max_num_locs 2 \
+        --pddl_easy_delivery_procedural_train_min_want_locs 1 \
+        --pddl_easy_delivery_procedural_train_max_want_locs 1 \
+        --pddl_easy_delivery_procedural_train_min_extra_newspapers 0 \
+        --pddl_easy_delivery_procedural_train_max_extra_newspapers 0 \
+        --pddl_easy_delivery_procedural_test_min_num_locs 2 \
+        --pddl_easy_delivery_procedural_test_max_num_locs 2 \
+        --pddl_easy_delivery_procedural_test_min_want_locs 1 \
+        --pddl_easy_delivery_procedural_test_max_want_locs 1 \
+        --pddl_easy_delivery_procedural_test_min_extra_newspapers 0 \
+        --pddl_easy_delivery_procedural_test_max_extra_newspapers 0 \
+        --num_train_tasks 5 \
+        --num_test_tasks 10 \
+        --debug
+"""
+from __future__ import annotations
+
+import logging
+from typing import Collection, Dict, FrozenSet, Iterator, List, Optional, \
+    Sequence, Set, Tuple
+
+from predicators.src import utils
+from predicators.src.approaches.open_loop_llm_approach import \
+    OpenLoopLLMApproach
+from predicators.src.llm_interface import OpenAILLM
+from predicators.src.planning import task_plan_with_option_plan_constraint
+from predicators.src.settings import CFG
+from predicators.src.structs import Box, Dataset, GroundAtom, Object, \
+    ParameterizedOption, Predicate, State, Task, Type, _GroundNSRT, _Option
+from predicators.src.utils import apply_operator, \
```

Comment:
for consistency with elsewhere, let's not import functions from utils, but rather just do `utils.apply_operator`, etc., in the code below